crates/supervisor/storage/src/providers/mod.rs

```diff
@@ -0,0 +1,5 @@
+mod derivation_provider;
+
+// todo: remove this when the provider is used
+#[allow(unused_imports)]
```

Comment:
```suggestion
#[expect(unused_imports)]
```
also here expect is helpful